user-management/src/test/java/de/conciso/keycloak/user/mgmt/ListenerITest.java

```diff
@@ -45,91 +40,64 @@ public class ListenerITest {
-        loginWithUserAccount();
-        loginWithAdminAccount();
-        extractAttributeValue();
-        assertNotNull(actualTimeValue);
-    }
-
     @Test
-    @Order(3)
     public void assertLastSuccessfulLoginAtAttributeMatchesExpected() {
-        extractLastSuccessfullyLoginTimeFromAccessToken();
-        long diff = calculateDifferenceInSecondsBetweenInstants(actualTimeValue, expectedTimeValue);
-        assertTrue(diff <= 1,
-            String.format("Expected time <%s> and actual time <%s> differ by more than 1 second.", expectedTimeValue, actualTimeValue));
-    }
-
-    @Test
-    @Order(4)
-    public void assertLastSuccessfulLoginAtAttributeIsUpToDate() throws InterruptedException {
-        Thread.sleep(3000);
-        loginWithUserAccount();
-        loginWithAdminAccount();
-        extractAttributeValue();
-        extractLastSuccessfullyLoginTimeFromAccessToken();
+        String authUrl = keycloak.getAuthServerUrl();
+        String userAccessToken = loginWithUserAccount(authUrl);
+        Instant expectedTimeValue = extractLastSuccessfullyLoginTimeFromAccessToken(userAccessToken);
+        Keycloak keycloakAdminClient = loginWithAdminAccount(authUrl);
+        Instant actualTimeValue = extractAttributeValue(keycloakAdminClient);
         long diff = calculateDifferenceInSecondsBetweenInstants(actualTimeValue, expectedTimeValue);
         assertTrue(diff <= 1,
             String.format("Expected time <%s> and actual time <%s> differ by more than 1 second.", expectedTimeValue, actualTimeValue));
     }
 
-    public void loginWithUserAccount() {
+    public String loginWithUserAccount(String url) {
         try {
             Keycloak keycloakUserClient = KeycloakBuilder.builder()
-                .serverUrl(authUrl)
+                .serverUrl(url)
                 .realm("conciso")
                 .username("test")
                 .password("test")
                 .clientId("admin-cli")
                 .build();
-            userAccessToken = keycloakUserClient.tokenManager().getAccessToken().getToken();
+            return keycloakUserClient.tokenManager().getAccessToken().getToken();
         } catch (Exception e) {
             throw new UnauthorizedException("User Authentication Failed.");
         }
     }
 
-    public void loginWithAdminAccount() {
+    public Keycloak loginWithAdminAccount(String url) {
         try {
-            keycloakAdminClient = KeycloakBuilder.builder()
-                .serverUrl(authUrl)
+            Keycloak adminClient = KeycloakBuilder.builder()
+                .serverUrl(url)
                 .realm("master")
                 .username(ADMIN_USER)
                 .password(ADMIN_PASS)
                 .clientId("admin-cli")
                 .build();
-            String accessToken = keycloakAdminClient.tokenManager().getAccessToken().getToken();
+            String accessToken = adminClient.tokenManager().getAccessToken().getToken();
```

Comment:
This call is not needed. Please remove. The adminClient will request a token automatically when you interact with the resources.